crates/bevy_audio/src/lib.rs

```diff
@@ -118,7 +118,11 @@ impl AddAudioSource for App {
     {
         self.init_asset::<T>().add_systems(
             PostUpdate,
-            (play_queued_audio_system::<T>, cleanup_finished_audio::<T>)
+            (
+                update_playing_audio_volume.run_if(resource_changed::<GlobalVolume>),
+                play_queued_audio_system::<T>,
+                cleanup_finished_audio::<T>,
+            )
                 .in_set(AudioPlaybackSystems),
         );
```

Comment:
This is adding a separate system for each type of audio source, which isn't needed. You can move this up into `AudioPlugin`'s `fn build`.